collections/_howtos/fixing_bugs.md

```diff
@@ -0,0 +1,146 @@
+
+```
+Refinement failed:
+Too few reflections to parameterise Crystal8Phi1, Crystal8Phi2, Crystal8Phi3, Crystal8g_param_0, Crystal8g_param_1.
+Try modifying refinement.parameterisation.auto_reduction options
+```
+
+Which suggests that indexing may have taken place, but not refinement. Quick jiffy script confirms that something is wrong:
+
+```
+from dials.array_family import flex
+from dxtbx.model.experiment_list import ExperimentList
+
+refl = flex.reflection_table.from_file("indexed.refl")
+expt = ExperimentList.from_file("indexed.expt")
+
+ids = set(refl["id"]) - set([-1])
+
+for i in ids:
+    e = expt[i]
+```
+
+=> create an issue:
+
+https://github.com/dials/dials/issues/1349
+
+This includes the input data, a script to reproduce the issue and this
+script to verify the problem.
+
+## The Fix
+Make some changes to fix the issue, identifying another potential issue along the way:
+
+https://github.com/dials/dials/issues/1351
+
+and 
+
+https://github.com/dials/dials/pull/1350
+
+## The Test Data
+
+Now this pull request may fix the issue but it would be good to ensure that it does not come back - so the data included in the example are uploaded to the dials-data-files repo:
+
+https://github.com/dials/data-files/commit/2e890287a925b69a3b5a55284166f71c4d173978
+
+Since they are really small this is fine and efficient, then added to the dials-data repo:
+
+https://github.com/dials/data/pull/210
+
+This includes a bunch of information about the data so people can see what it’s about and where it came from - this data was my data so was at liberty to make it public. This is the included information, in a yaml file:
+
+```
+name: Multiple lattices for indexing stills, from VMXi thaumatin grid scan
+author: Graeme Winter (2019)
+license: CC-BY 4.0
+url: https://ispyb.diamond.ac.uk/dc/visit/nt24686-7/id/3906663
+description: >
+  Experiment and reflection file data corresponding to image 07603 (i.e.
+  split 07602) from a position in a thaumatin grid scan where multiple
+  crystal lattices overlap. 
+
+data:
+ - url: https://github.com/dials/data-files/raw/2e890287a925b69a3b5a55284166f71c4d173978/vmxi_thaumatin_grid_index/split_07602.expt
+ - url: https://github.com/dials/data-files/raw/2e890287a925b69a3b5a55284166f71c4d173978/vmxi_thaumatin_grid_index/split_07602.refl
+```
+
+There was much to-and-fro with this as you make a PR, then merge it, then automated systems take over to annotate it and publish an update to dials-data. So after that, 
+
+pip install -U dials_data
+
+gives you access to the new files.
+
+## The Test
+
+These data can then be used to add a regression test to the PR, so not only has a bug been found and fixed, but it prevents it from coming back - this is essentially codifying the information in the original issue:
+
+```
+from dxtbx.model.experiment_list import ExperimentList
+from dials.array_family import flex
+import procrunner
+
+
+def test_all_expt_ids_have_expts(dials_data, tmpdir):
```

Comment:
just a note: We should be moving away from `tmpdir` towards `tmp_path` for future code. This is not quite a drop-in replacement and works slightly differently, but it is the standard that python settled on.

Reply:
Added note to the docs, but not changed for consistency with other tests in the file.